src/Pact/Typechecker.hs

```diff
@@ -1193,7 +1193,13 @@ singLens = iso pure head
 
 -- | Typecheck a top-level production.
 typecheck :: TopLevel Node -> TC (TopLevel Node)
-typecheck f@(TopFun FDefun {} _) = typecheckBody f (tlFun . fBody)
+typecheck f@(TopFun FDefun {} _) = typecheckBody f (tlFun . fBody) >>= \case
+  TopFun fd@FDefun{} i |  x:_ <- reverse $ _fBody fd ->
```

Comment:
Just a personal nit, but I'm not a fan of using `$` like this. I like `$` when it separates domains, like `put $ _fBody fd`, but not when it's used for pure arguments to a pure function. `$ ` (dollar and a space) is no space savings over `()`....

Reply:
Fair argument, I try to adopt this guideline